### PR TITLE
MITAB: Expose font family to feature style symbol ID

### PR DESCRIPTION
**Pull request by drons**
This PR expose font family as additional style ID.

MapInfo Interchange Format (mif/mid) provides symbol style based on font, but now it is ignored when exporting style from TABFontPoint::GetStyleString() [(see: Appendix J: MapInfo Data Interchange Format, Symbol Clause – TrueType Font Syntax section)](https://web.archive.org/web/20060909054240/http://resource.mapinfo.com/static/files/document/1074660800077/interchange_file.pdf)

Previously, a font field was provided by OGRStyleSymbol, but it was deleted in b98fd4b51bda2f5de0b8c5876b704e5a2fa3ca95.
If it could be brought back, in my opinion, it would be a good place to store the symbol font.

### CODE REVIEW

**Comment by drons** [comment]
I will return the code associated with OGRSTSymbolFontName in the next commit, and redo my PR.

**Comment by rouault** [comment]
yes please update ogr_feature_style.html (and rewrite history to cancel your first commit and its reversal)

**Comment by drons** [comment]
I did what I wanted. 
But now for font symbols, the ID in the form of `"mapinfo-sym-%d,ogr-sym-%d"` looks unusual. It may be to add a specialization for TABFontPoint :: GetSymbolStyleString (), which will return a style with an ID in the form `"font-sym-%d,ogr-sym-9"`?

**Comment by rouault** [comment]
> But now for font symbols, the ID in the form of `"mapinfo-sym-%d,ogr-sym-%d"` looks unusual. It may be to add a specialization for TABFontPoint :: GetSymbolStyleString (), which will return a style with an ID in the form `"font-sym-%d,ogr-sym-9"`?

I'm afraid I didn't get what you mean

**Comment by drons** [comment]
Take a look. 
All MITAB point symbols objects call [ITABFeatureSymbol::GetSymbolStyleString](https://github.com/OSGeo/gdal/blob/master/gdal/ogr/ogrsf_frmts/mitab/mitab_feature.cpp#L8848) to export style as OGR string.
This function create symbol ID in form `"mapinfo-sym-%d,ogr-sym-%d"`.
For a font based symbols it is strange for me to use ID in form `mapinfo-*`.
In this case, I propose to replace `mapinfo-sym-%d`  with `font-sym-%d`, where %d is a glyph number inside a font.
`ogr-sym-9` (star - default MapInfo symbol style) - is a fallback symbol ID because we [should](https://github.com/OSGeo/gdal/blob/master/gdal/ogr/ogr_feature_style.html#L580) always include one of the OGR ids at the end of the comma-delimited list of ids.

**Comment by rouault** [comment]
> In this case, I propose to replace `mapinfo-sym-%d` with `font-sym-%d`, where %d is a glyph number inside a font.

ok, makes sense now. Sounds reasonable.

**Comment by rouault** [comment]
I guess we can merge as it, and the tuning for point symbol object can be done in a later PR ? As you wish

File: gdal/ogr/ogrsf_frmts/mitab/mitab_feature.cpp [hunk 1]
```diff
@@ -1558,6 +1558,27 @@ const char *TABFontPoint::GetStyleString() const
...
+{
+    ITABFeatureSymbol::SetSymbolFromStyle(poSymbolStyle);
+
+    GBool bIsNull = 0;
+
+    // Try to set font glyph number
+    const char* pszSymbolId = poSymbolStyle->Id(bIsNull);
+    if((!bIsNull) && pszSymbolId && strstr(pszSymbolId, "font-sym-"))
+    {
+        const int nSymbolId = atoi(pszSymbolId+9);
```
Comment (rouault) [inline gdal/ogr/ogrsf_frmts/mitab/mitab_feature.cpp]: this assumes that the pszSymbolId starts with "font-sym-", but strstr() tests only for its existence. So either use STARTS_WITH(), or take the pointer returned by strstr()

**Comment by drons** [comment]
Thanks!

**Pull request merged**